### PR TITLE
Add chosen mitigation of event with multiple choices to graph datalabels

### PR DESCRIPTION
**Pull request by Petr26**
Screen Shot 2021-01-13 at 4 14 33 PM

In case I set mitigations while event overlay is active then in graph is shown only Event-like datalabel with details in tooltip.
Screen Shot 2021-01-13 at 4 14 30 PM

### CODE REVIEW

File: frontend/src/app/game/events-layout/events-layout.component.ts [hunk 1]
```diff
@@ -15,12 +15,13 @@ export class EventsLayoutComponent {
   constructor(public gameService: GameService) {
   }
 
-  resumeEvent(eventMitigation?: EventMitigation) {
-    if (eventMitigation !== undefined) {
-      this.gameService.game.applyMitigationActions({eventMitigations: [eventMitigation]});
+  resumeEvent(event: Event, chosenMitigation: EventMitigation | undefined) {
```
Comment (jnwltr) [inline frontend/src/app/game/events-layout/events-layout.component.ts:18]: You can do this using optional (`?`) param, or a default value. In either case when invoked, `undefined` arg is not needed.

File: frontend/src/app/game/game.service.ts [hunk 2]
```diff
@@ -30,8 +35,20 @@ export class GameService {
...
+
+    this.activateEvent$.pipe(
+      filter(activatedEvent => {
+        const mitigations = activatedEvent?.originEvent.mitigations;
+        return Boolean(mitigations && mitigations.length > 1);
+      }),
+      tap(activatedEvent => this.activatedEvent = activatedEvent),
+      debounceTime(this.PLAY_SPEED),
+      untilDestroyed(this),
+    ).subscribe(() => this.activatedEvent = undefined);
```
Comment (jnwltr) [inline frontend/src/app/game/game.service.ts:51]: This construct is pretty cryptic and error-prone, pls, could you improve it, pls?

File: frontend/src/app/game/graphs/line-graph/line-graph.component.ts [hunk 3]
```diff
@@ -104,20 +131,30 @@ export class LineGraphComponent implements OnInit, AfterViewInit {
...
+          let datalabel = this.dataLabelNodes[index]?.mitigations;
+          if (this.dataLabelNodes[index]?.event) {
+            datalabel = this.dataLabelNodes[index]?.event?.chosenMitigation?.label;
+          }
+          return datalabel;
+        },
         font: context => {
           const width = context.chart.width;
-          const size = Math.round(width! / 64);
+          let size = Math.round(width! / 64);
```
Comment (jnwltr) [inline frontend/src/app/game/graphs/line-graph/line-graph.component.ts:156]: Where did `64` come from?

**Review by jnwltr — CHANGES_REQUESTED** [review]
Pls, show us a screenshot, and check my comments.

File: frontend/src/app/game/game.service.ts (hunk 2 above)
Comment (Petr26) [reply frontend/src/app/game/game.service.ts:51]: Idea is to once tick active event and remove it after "game speed" time passes. Any idea how to do it?

File: frontend/src/app/game/graphs/line-graph/line-graph.component.ts (hunk 3 above)
Comment (Petr26) [reply frontend/src/app/game/graphs/line-graph/line-graph.component.ts:156]: Only in my head, tested on various resolutions. Looks good.

File: frontend/src/app/game/game.service.ts (hunk 2 above)
Comment (Petr26) [reply frontend/src/app/game/game.service.ts]: totally removed and replaced with simple variable that is reset in `tick` method.

**Review by jnwltr — APPROVED** [review]

**Pull request merged**